client/volume_inspect_test.go

```diff
@@ -0,0 +1,67 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/docker/engine-api/client/transport"
+	"github.com/docker/engine-api/types"
+)
+
+func TestVolumeInspectError(t *testing.T) {
+	client := &Client{
+		transport: transport.NewMockClient(nil, transport.ErrorMock(http.StatusInternalServerError, "Server error")),
+	}
+
+	_, err := client.VolumeInspect("nothing")
+	if err == nil || err.Error() != "Error response from daemon: Server error" {
+		t.Fatalf("expected a Server Error, got %v", err)
+	}
+}
+
+func TestVolumeInspectNotFound(t *testing.T) {
+	client := &Client{
+		transport: transport.NewMockClient(nil, transport.ErrorMock(http.StatusNotFound, "Server error")),
+	}
+
+	_, err := client.VolumeInspect("unknown")
+	if err == nil || !IsErrVolumeNotFound(err) {
+		t.Fatalf("expected a volumeNotFound error, got %v", err)
+	}
+}
+
+func TestVolumeInspect(t *testing.T) {
+	client := &Client{
+		transport: transport.NewMockClient(nil, func(req *http.Request) (*http.Response, error) {
+			content, err := json.Marshal(types.Volume{
```

Comment:
Hum I think I almost never did for `GET` methods, but no valid reason not to do it :stuck_out_tongue_closed_eyes:. Will update :wink: